src/NITTA/Model/Microarchitecture/Builder.hs

```diff
@@ -42,26 +43,33 @@ data BuilderSt tag v x t = BuilderSt
     { signalBusWidth :: Int
     , availPorts :: [SignalTag]
     , puProtos :: [(tag, PU v x t)]
-    , netEnv :: UnitEnv (BusNetwork tag v x t)
     }
 
+netIOPorts puProtos =
+    let setOfPorts ports = foldl (flip $ S.union . ports . snd) S.empty puProtos
```

Comment:
Will be better if you move it where section. 

Also, you can rewrite it with unionMapf from project utils.